.jenkins/pipelines/standalone/report-code-coverage.Jenkinsfile

```diff
@@ -0,0 +1,83 @@
+pipeline {
+    agent {
+        label 'Jenkins-Shared-DC2'
+    }
+    options {
+        timeout(time: 30, unit: 'MINUTES')
+    }
+    parameters {
+        string(name: "REPOSITORY", defaultValue: "deislabs")
+        string(name: "BRANCH", defaultValue: "main", description: "Branch to build")
+        string(name: "COMMIT_ID", description: "Short commit ID used to archive build resoures")
+        string(name: "BUILD_URL", description: "URL of the upstream build")
+    }
+    environment {
+        MYST_SCRIPTS =    "${WORKSPACE}/scripts"
+        JENKINS_SCRIPTS = "${WORKSPACE}/.jenkins/scripts"
+        LCOV_FILE =       "lcov-${COMMIT_ID}.info"
+        BUILD_INFO =      "${BUILD_URL == '' ? 'N/A' : BUILD_URL}"
+    }
+    stages {
+        stage('Init Config') {
+            steps {
+                sh """
+                   # Initialize dependencies repo
+                   ${JENKINS_SCRIPTS}/global/init-config.sh
+
+                   # Install global dependencies
+                   ${JENKINS_SCRIPTS}/global/wait-dpkg.sh
+                   ${JENKINS_SCRIPTS}/global/init-install.sh
+                   """
+            }
+        }
+        stage('Download code coverage report') {
+            steps {
+                azureDownload(
+                    downloadType: 'container',
+                    containerName: 'mystikos-code-coverage',
+                    includeFilesPattern: "${LCOV_FILE}",
+                    storageCredentialId: 'mystikosreleaseblobcontainer'
+                )
+            }
+        }
+        stage('Report code coverage') {
+            when {
+                expression { return fileExists(LCOV_FILE) }
+            }
+            steps {
+                sh """
+                   lcov --list ${LCOV_FILE} | tee code-coverage-summary
+
+                   tail -n 1 code-coverage-summary | sed 's/|/ /g' | awk '{print "[Line: " \$2 ", Function: " \$4 "] Code Coverage Report"}' > email-subject
+
+                   tail -n 1 code-coverage-summary | sed 's/|/ /g' | awk '{print "Summary:\\nLine coverage: " \$2 ", function coverage: " \$4 "\\n\\nBuild URL: ${BUILD_INFO}\\n\\nFull Report:"}' >> email-body
+                   cat code-coverage-summary >> email-body
+                   """
+
+                script {
+                    LCOV_REPORT = sh(
+                        returnStdout: true,
+                        script: "cat email-body"
+                    )
+                    EMAIL_SUBJECT = sh(
+                        returnStdout: true,
+                        script: "echo \"\$(cat email-subject) (\$(date '+%Y-%m-%d'))\""
+                    )
+
+                    build job:"Send-Email",
```

Comment:
If either of the `Measure Code Coverage` or `Report Code Coverage` stage fails for any reason, will an email still be sent out?

Reply:
There won't be, I'll configure it so that a failure email will be sent out

Reply:
I configured the `Report Code Coverage` step to send a failure email if the lcov file doesn't exist.